homeassistant/components/mqtt/__init__.py

```diff
@@ -122,6 +125,8 @@ def valid_discovery_topic(value):
         vol.Inclusive(CONF_CLIENT_CERT, 'client_key_auth',
                       msg=CLIENT_KEY_AUTH_MSG): cv.isfile,
         vol.Optional(CONF_TLS_INSECURE): cv.boolean,
+        vol.Optional(CONF_TLS_VERSION,
+                     default=DEFAULT_TLS_PROTOCOL): cv.string,
```

Comment:
Instead of a string, we should validate only valid options: `vol.Any('1.0', '1.1', '1.2')`. You can leave the default out as then the default is None which will result in the auto behavior.